test/unit/mdc-slider/foundation.test.js

```diff
@@ -135,6 +151,94 @@ test('#destroy deregisters all component event handlers registered during init()
   td.verify(mockAdapter.deregisterResizeHandler(isA(Function)));
 });
 
+test('#setupTrackMarker append right number of markers to discrete slider with markers', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  td.when(mockAdapter.hasClass(cssClasses.IS_DISCRETE)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.HAS_TRACK_MARKER)).thenReturn(true);
+  foundation.init();
+  raf.flush();
+
+  const numMarkers = 10;
+  foundation.setMax(100);
+  foundation.setMin(0);
+  foundation.setStep(10);
+  foundation.setupTrackMarker();
+
+  td.verify(mockAdapter.removeTrackMarkers());
+  td.verify(mockAdapter.appendTrackMarkers(numMarkers));
+
+  raf.restore();
+});
+
+test('#setupTrackMarker append one excessive marker if distance is indivisible to step', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  td.when(mockAdapter.hasClass(cssClasses.IS_DISCRETE)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.HAS_TRACK_MARKER)).thenReturn(true);
+  foundation.init();
+  raf.flush();
+
+  const numMarkers = 12;
+  foundation.setMax(100);
+  foundation.setMin(0);
+  foundation.setStep(9);
+  foundation.setupTrackMarker();
+
+  td.verify(mockAdapter.removeTrackMarkers());
+  td.verify(mockAdapter.appendTrackMarkers(numMarkers));
+
+  raf.restore();
+});
+
+test('#setupTrackMarker does execute if it is continuous slider', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  td.when(mockAdapter.hasClass(cssClasses.IS_DISCRETE)).thenReturn(false);
+  td.when(mockAdapter.hasClass(cssClasses.HAS_TRACK_MARKER)).thenReturn(true);
+  foundation.init();
+  raf.flush();
+
+  foundation.setMax(100);
+  foundation.setMin(0);
+  foundation.setStep(10);
+  foundation.setupTrackMarker();
+
+  td.verify(mockAdapter.removeTrackMarkers(), {times: 0});
+  td.verify(mockAdapter.appendTrackMarkers(isA(Number)), {times: 0});
+
+  raf.restore();
+});
+
+test('#setupTrackMarker does execute if discrete slider does not display markers', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  td.when(mockAdapter.hasClass(cssClasses.IS_DISCRETE)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.HAS_TRACK_MARKER)).thenReturn(false);
+  foundation.init();
+  raf.flush();
+
+  foundation.setMax(100);
+  foundation.setMin(0);
+  foundation.setStep(10);
+  foundation.setupTrackMarker();
```

Comment:
Yes sir, 4 tests below.

`#setupTrackMarker does execute if discrete slider does not display markers`